examples/dialog_async/main.rs

```diff
@@ -36,30 +20,41 @@ fn build_ui(application: &gtk::Application) {
         .visible(true)
         .build();
 
-    let window = Rc::new(
```

Comment:
The `Rc` here was very strange

Reply:
Yep...

I also found the image holder in cairo_threads a bit too over-engineered.
Maybe it should also be simplified
https://github.com/gtk-rs/gtk-rs/blob/master/examples/cairo_threads/image/image_holder.rs

Reply:
To be more explicit, remove the image_holder altogether and wrap `Image.data` in a Rc<Refcell<>>
https://github.com/gtk-rs/gtk-rs/blob/master/examples/cairo_threads/image/mod.rs

Reply:
Can you show what you're thinking of?

Reply:
Sure will open another PR